### PR TITLE
Remove connection manager

### PR DESCRIPTION
**Pull request by bording**
This is the same as #152, only opened against the correct branch.

I believe this addresses all the feedback from #152, so this should be ready for final review. Once the review is done, I'm thinking it makes sense to squash this down.

Connects to #149

### CODE REVIEW

File: src/NServiceBus.RabbitMQ/Connection/ChannelProvider.cs [hunk 1]
```diff
@@ -45,21 +41,20 @@ public void ReturnPublishChannel(ConfirmsAwareChannel channel)
...
         {
-            lock (connectionFactory)
-            {
-                connection = connection ?? connectionFactory.CreatePublishConnection();
-            }
+            //injected
         }
 
-        public void Dispose()
+        void DisposeManaged()
```
Comment (adamralph) [inline src/NServiceBus.RabbitMQ/Connection/ChannelProvider.cs]: Is this some kind of Janitor magic?

**Comment by adamralph** [comment]
@bording the changes LGTM.

Happy to merge once you squash.

File: src/NServiceBus.RabbitMQ/Connection/ChannelProvider.cs [hunk 2]
```diff
@@ -38,7 +41,20 @@ public void ReturnPublishChannel(ConfirmsAwareChannel channel)
             }
         }
 
-        readonly ConnectionManager connectionManager;
+        public void Dispose()
+        {
+            //injected
+        }
+
+        void DisposeManaged()
```
Comment (adamralph) [inline src/NServiceBus.RabbitMQ/Connection/ChannelProvider.cs]: Is this some kind of Janitor magic?

Comment (bording) [inline src/NServiceBus.RabbitMQ/Connection/ChannelProvider.cs]: Yes, that's how Janitor lets you hook into the generated method.

**Comment by bording** [comment]
Squashed!

**Comment by adamralph** [comment]
Thanks @bording!

**Pull request merged**